Material UI Select widget label always shrinked

The `SelectWidget` always uses a shrinked label:
https://github.com/rjsf-team/react-jsonschema-form/blob/v5.0.0-beta.16/packages/material-ui/src/SelectWidget/SelectWidget.tsx#L67

The `shrink` property should be dependent on the current value, whether it is empty or not

I can provide a PR if it make sense.

ok. I have forked the repo and trying to build `validator-ajv8` I get this error, any idea? I just forked, npm install, and trying to build...
```
(typescript) Error: D:/MyProjects/react-jsonschema-form/packages/validator-ajv8/src/createAjvInstance.ts(45,16): semantic error TS2345: Argument of type 'import("D:/MyProjects/react-jsonschema-form/packages/validator-ajv8/node_modules/ajv8/dist/ajv").default' is not assignable to parameter of type 'import("D:/MyProjects/react-jsonschema-form/packages/validator-ajv8/node_modules/ajv/dist/core").default'.
  Types of property 'opts' are incompatible.
    Type 'import("D:/MyProjects/react-jsonschema-form/packages/validator-ajv8/node_modules/ajv8/dist/core").InstanceOptions' is not assignable to type 'import("D:/MyProjects/react-jsonschema-form/packages/validator-ajv8/node_modules/ajv/dist/core").InstanceOptions'.
      Type 'InstanceOptions' is not assignable to type 'CurrentOptions'.
        Types of property 'keywords' are incompatible.
          Type 'import("D:/MyProjects/react-jsonschema-form/packages/validator-ajv8/node_modules/ajv8/dist/types/index").Vocabulary | undefined' is not assignable to type 'import("D:/MyProjects/react-jsonschema-form/packages/validator-ajv8/node_modules/ajv/dist/types/index").Vocabulary | undefined'.
            Type 'Vocabulary' is not assignable to type '(string | KeywordDefinition)[]'.
              Type 'string | import("D:/MyProjects/react-jsonschema-form/packages/validator-ajv8/node_modules/ajv8/dist/types/index").KeywordDefinition' is not assignable to type 'string | import("D:/MyProjects/react-jsonschema-form/packages/validator-ajv8/node_modules/ajv/dist/types/index").KeywordDefinition'.
                Type 'CodeKeywordDefinition' is not assignable to type 'string | KeywordDefinition'.
                  Type 'import("D:/MyProjects/react-jsonschema-form/packages/validator-ajv8/node_modules/ajv8/dist/types/index").CodeKeywordDefinition' is not assignable to type 'import("D:/MyProjects/react-jsonschema-form/packages/validator-ajv8/node_modules/ajv/dist/types/index").CodeKeywordDefinition'.
                    Types of property 'code' are incompatible.
                      Type '(cxt: import("D:/MyProjects/react-jsonschema-form/packages/validator-ajv8/node_modules/ajv8/dist/compile/validate/index").KeywordCxt, ruleType?: string | undefined) => void' is not assignable to type '(cxt: import("D:/MyProjects/react-jsonschema-form/packages/validator-ajv8/node_modules/ajv/dist/compile/validate/index").KeywordCxt, ruleType?: string | undefined) => void'.
                        Types of parameters 'cxt' and 'cxt' are incompatible.
                          Type 'import("D:/MyProjects/react-jsonschema-form/packages/validator-ajv8/node_modules/ajv/dist/compile/validate/index").KeywordCxt' is not assignable to type 'import("D:/MyProjects/react-jsonschema-form/packages/validator-ajv8/node_modules/ajv8/dist/compile/validate/index").KeywordCxt'.
                            The types of 'gen._scope' are incompatible between these types.
                              Type 'import("D:/MyProjects/react-jsonschema-form/packages/validator-ajv8/node_modules/ajv/dist/compile/codegen/scope").Scope' is not assignable to type 'import("D:/MyProjects/react-jsonschema-form/packages/validator-ajv8/node_modules/ajv8/dist/compile/codegen/scope").Scope'.
                                Property '_names' is protected but type 'Scope' is not a class derived from 'Scope'.
```

--- a/packages/material-ui/src/SelectWidget/SelectWidget.tsx
+++ b/packages/material-ui/src/SelectWidget/SelectWidget.tsx
@@ -43,7 +43,13 @@ export default function SelectWidget<
 }: WidgetProps<T, S, F>) {
   const { enumOptions, enumDisabled } = options;
 
+  multiple = typeof multiple === "undefined" ? false : !!multiple;
+
   const emptyValue = multiple ? [] : "";
+  const isEmpty =
+    typeof value === "undefined" ||
+    (multiple && value.length < 1) ||
+    (!multiple && value === emptyValue);
 
   const _onChange = ({
     target: { value },
@@ -61,7 +67,7 @@ export default function SelectWidget<
       id={id}
       name={id}
       label={label || schema.title}
-      value={typeof value === "undefined" ? emptyValue : value}
+      value={isEmpty ? emptyValue : value}
       required={required}
       disabled={disabled || readonly}
       autoFocus={autofocus}
@@ -74,11 +80,11 @@ export default function SelectWidget<
       select // Apply this and the following props after the potential overrides defined in textFieldProps
       InputLabelProps={{
         ...textFieldProps.InputLabelProps,
-        shrink: true,
+        shrink: !isEmpty,
       }}
       SelectProps={{
         ...textFieldProps.SelectProps,
-        multiple: typeof multiple === "undefined" ? false : multiple,
+        multiple,
       }}
       aria-describedby={ariaDescribedByIds<T>(id)}
     >

--- a/packages/mui/src/SelectWidget/SelectWidget.tsx
+++ b/packages/mui/src/SelectWidget/SelectWidget.tsx
@@ -43,7 +43,13 @@ export default function SelectWidget<
 }: WidgetProps<T, S, F>) {
   const { enumOptions, enumDisabled } = options;
 
+  multiple = typeof multiple === "undefined" ? false : !!multiple;
+
   const emptyValue = multiple ? [] : "";
+  const isEmpty =
+    typeof value === "undefined" ||
+    (multiple && value.length < 1) ||
+    (!multiple && value === emptyValue);
 
   const _onChange = ({
     target: { value },
@@ -61,7 +67,7 @@ export default function SelectWidget<
       id={id}
       name={id}
       label={label || schema.title}
-      value={typeof value === "undefined" ? emptyValue : value}
+      value={isEmpty ? emptyValue : value}
       required={required}
       disabled={disabled || readonly}
       autoFocus={autofocus}
@@ -74,11 +80,11 @@ export default function SelectWidget<
       select // Apply this and the following props after the potential overrides defined in textFieldProps
       InputLabelProps={{
         ...textFieldProps.InputLabelProps,
-        shrink: true,
+        shrink: !isEmpty,
       }}
       SelectProps={{
         ...textFieldProps.SelectProps,
-        multiple: typeof multiple === "undefined" ? false : multiple,
+        multiple,
       }}
       aria-describedby={ariaDescribedByIds<T>(id)}
     >

